From 5cf0b52d291c81d3a6b2a3082f4ee7845e75e635 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 9 May 2014 23:59:19 +0200
Subject: [PATCH 62/70] MEDIUM: checks: only complain about the missing port
 when the check uses TCP

For UNIX socket addresses, we don't need any port, so let's disable the
check under this condition.
---
 src/server.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/server.c b/src/server.c
index 9fa69f2..1901686 100644
--- a/src/server.c
+++ b/src/server.c
@@ -1072,9 +1072,11 @@ int parse_server(const char *file, int linenum, char **args, struct proxy *curpr
 			}
 			/*
 			 * We need at least a service port, a check port or the first tcp-check rule must
-			 * be a 'connect' one
+			 * be a 'connect' one when checking an IPv4/IPv6 server.
 			 */
-			if (!newsrv->check.port) {
+			if (!newsrv->check.port &&
+			    (is_inet_addr(&newsrv->check_common.addr) ||
+			     (!is_addr(&newsrv->check_common.addr) && is_inet_addr(&newsrv->addr)))) {
 				struct tcpcheck_rule *n = NULL, *r = NULL;
 				struct list *l;
 
-- 
1.8.3.2

